shared/interceptor/transform.interceptor.ts

```diff
@@ -0,0 +1,42 @@
+import { CallHandler, ExecutionContext, Injectable, NestInterceptor } from '@nestjs/common';
+import { Reflector } from '@nestjs/core';
+import { responseMessageKey } from '@shared/decorator/set-response.decorator';
+import { map, Observable } from 'rxjs';
+
+interface CustomRes<T> {
+  status: number;
+  data: T;
+  message: string;
+}
+
+@Injectable()
+export class TransformInterceptor<T> implements NestInterceptor<T, CustomRes<T>> {
+  public constructor(private reflector: Reflector) {}
+
+  public intercept(
+    context: ExecutionContext,
+    next: CallHandler<T>,
+  ): Observable<CustomRes<T>> | Promise<Observable<CustomRes<T>>> {
+    const response = this.reflector.get<{ message: string; status: number }>(
+      responseMessageKey,
+      context.getHandler(),
+    );
+
+    const status = response.status
+      ? response.status
+      : context.switchToHttp().getResponse().statusCode;
```

Comment:
요렇게 표현해줘도 괜찮을 것 같습니다~
```suggestion
    const status = response.status ?? context.switchToHttp().getResponse().statusCode;
```

Reply:
[907890d](https://github.com/GrebbMe/grabbme-be/pull/45/commits/907890d8fa731a3e7a1db51e9ea2d2f30c08aa1b)

해당 커밋에 말씀해주신부분 적용하여 수정 진행 했습니다!